### PR TITLE
[WFLY-8334] Make it possible to set username and password when Elytro…

### PR DESCRIPTION
**Pull request by fl4via**
…n is enabled

Jira: https://issues.jboss.org/browse/WFLY-8334
EAP Jira: https://issues.jboss.org/browse/JBEAP-9327

### CODE REVIEW

**Comment by fl4via** [comment]
EAP7 PR: https://github.com/jbossas/jboss-eap7/pull/1477

File: connector/src/main/java/org/jboss/as/connector/subsystems/datasources/Constants.java [hunk 1]
```diff
@@ -457,7 +457,7 @@ public void validateResolvedParameter(String parameterName, ModelNode value) thr
             .setXmlName(Security.Tag.AUTHENTICATION_CONTEXT.getLocalName())
             .setAllowExpression(true)
             .setRequires(ELYTRON_ENABLED_NAME)
-            .addAlternatives(SECURITY_DOMAIN_NAME, USERNAME_NAME)
+            .addAlternatives(SECURITY_DOMAIN_NAME)
```
Comment (simkam) [inline connector/src/main/java/org/jboss/as/connector/subsystems/datasources/Constants.java]: is this change necessary? it still shouldn't be allowed to set authentication-context and username together.

Comment (simkam) [inline connector/src/main/java/org/jboss/as/connector/subsystems/datasources/Constants.java]: nm, I see that is alternatives set on username attribute

.addAlternatives(SECURITY_DOMAIN_NAME, AUTHENTICATION_CONTEXT_NAME)

**Comment by fl4via** [comment]
@simkam You are right, I'll fix this, thanks for your review
